rs/ic_os/os_tools/hostos_tool/src/main.rs

```diff
@@ -73,24 +71,15 @@ pub fn main() -> Result<()> {
                 ))?;
             eprintln!("Deployment config: {:?}", deployment_settings);
 
-            let mgmt_mac = match deployment_settings.deployment.mgmt_mac {
-                Some(config_mac) => {
-                    let mgmt_mac = FormattedMacAddress::try_from(config_mac.as_str())?;
-                    eprintln!(
-                        "Using mgmt_mac address found in deployment.json: {}",
-                        mgmt_mac
-                    );
-                    mgmt_mac
-                }
-                None => get_ipmi_mac()?,
-            };
-            let generated_mac = generate_mac_address(
-                &mgmt_mac,
-                deployment_settings.deployment.name.as_str(),
-                &NodeType::HostOS,
+            let mgmt_mac = resolve_mgmt_mac(deployment_settings.deployment.mgmt_mac)?;
+            let generated_mac = calculate_deterministic_mac(
+                mgmt_mac,
+                deployment_settings.deployment.name.parse()?,
+                IpVariant::V6,
+                0x0,
```

Comment:
What is 0x0? Not sure if we have a style guide for this, but one option is adding a arg-name comment, like `/* my_arg */ 0x0`. Or if 0x0 has a special meaning (HostOS?), we could extract it into a constant

Reply:
These `u8`s will also be factored out, but I've added comments in the meantime to keep things clear.